### PR TITLE
[ASCII-1079] Return all queryable config values when fetching bare config endpoint path

### PR DESCRIPTION
**Pull request by pgimalac**
### What does this PR do?

Return the map of all authorized config keys to their values in the config API when querying the bare endpoint (ie. without specifying any config key).

### Motivation

The endpoint only accepts a limited set of config as parameters, and returns a 403 otherwise.
Most of the time we'll just query all values anyway, doing so in a single request is easier and more efficient.

### Describe how to test/QA your changes

Unit tests ensure that the list is properly returned by the endpoint.

For manual testing, enable the config api:
```
agent_ipc_port: 5004
```

and run:
```
curl -L -k -H "authorization: Bearer $(cat <path_to_auth_token>)" https://localhost:5004/config/v1
```

It should all authorized keys along with their values as a JSON map, ie. `{"api_key": "qwertyuiop"}`.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by sgnn7** [comment]
@pgimalac I edited the PR header for clarity since it made it seem like we're dealing with access keys rather than config field keys

File: comp/api/api/apiimpl/internal/config/endpoint.go [hunk 1]
```diff
@@ -50,6 +50,25 @@ func (c *configEndpoint) serveHTTP(w http.ResponseWriter, r *http.Request) {
...
+		keys = append(keys, key)
+	}
+
+	body, err := json.Marshal(keys)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
```
Comment (sgnn7) [inline comp/api/api/apiimpl/internal/config/endpoint.go:65]: nit: `w.Write` writes the `http.StatusOK` header by default iirc but up to you if you want it explicitly listed

**Review by sgnn7 — COMMENTED** [review]
@pgimalac Couple of non-blocking comments

File: comp/api/api/apiimpl/internal/config/endpoint.go (hunk 1 above)
Comment (pgimalac) [reply comp/api/api/apiimpl/internal/config/endpoint.go:65]: I know but I'm not a big fan of implicit behavior, I find it clearer to have explicitly what status is used :smile:

File: comp/api/api/apiimpl/internal/config/endpoint.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ type configEndpoint struct {
 	errorsExpvar       expvar.Map
 }
 
-func (c *configEndpoint) serveHTTP(w http.ResponseWriter, r *http.Request) {
+func (c *configEndpoint) valueServeHTTP(w http.ResponseWriter, r *http.Request) {
```
[the remark this answers is not in the document]
Comment (pgimalac) [reply comp/api/api/apiimpl/internal/config/endpoint.go:39]: I clearly agree, I'm just terrible at naming things, I'll go with your suggestion !

[the remark this answers is not in the document]
Comment (pgimalac) [reply comp/api/api/apiimpl/internal/config/endpoint.go:39]: For the second I'll actually change what it does to return all config keys + values (not just keys) so I'll go with `getAllConfigValuesHandler`, let me know if you have a better name :smile:

**Review by ogaca-dd — APPROVED** [review]

File: comp/api/api/apiimpl/internal/config/endpoint.go [hunk 3]
```diff
@@ -36,18 +37,44 @@ type configEndpoint struct {
...
 		return
 	}
 
-	w.WriteHeader(statusCode)
-	_, err = w.Write(body)
-	if err != nil {
-		log.Warnf("config endpoint: could not write response body: %v", err)
+	log.Debug("config endpoint received a request from '%s' for config '%s'", r.RemoteAddr, path)
+	value := c.cfg.Get(path)
+	if value == nil {
```
Comment (hush-hush) [inline comp/api/api/apiimpl/internal/config/endpoint.go:56]: A setting could be `nil` as `null` is a valid YAML type. If we want to handle this case we should use the `IsSet` method on the configuration.
But should we care ? If something it not set, returning `nil`/`null` is a valid behavior and not really and error, no ?

**Review by hush-hush — CHANGES_REQUESTED** [review]

File: comp/api/api/apiimpl/internal/config/endpoint.go (hunk 3 above)
Comment (pgimalac) [reply comp/api/api/apiimpl/internal/config/endpoint.go:56]: For this specific change I kept the same behavior as in my previous PR, but thinking about it I agree `nil` could be a valid value so it doesn't really makes sense to ignore it, I'll update the PR

Comment (pgimalac) [reply comp/api/api/apiimpl/internal/config/endpoint.go]: I also removed the expvar about nil values since we consider it a "normal" value now

**Review by hush-hush — APPROVED** [review]

**Pull request merged**